### PR TITLE
Extended the FAQ

### PR DESCRIPTION
**Pull request by revans2**
The extends the FAQ to include more questions that we have seen come up.

### CODE REVIEW

**Comment by revans2** [comment]
build

File: docs/FAQ.md [hunk 1]
```diff
@@ -13,7 +13,174 @@ settings. This is true of all configs in Spark. If you changed
...
+Reference Architectures should be available around Q4 2020.
+
+### What CUDA versions are supported?
+
+CUDA 10.1 and 10.2 are currently supported, but you need to download the cudf jar that corresponds
+to the version you are using.
+
+### What parts of Apache Spark are accelerated?
+
+Currently, A limited set of SQL and DataFrame operations are supported, please see the
```
Comment (kuhushukla) [inline docs/FAQ.md:49]: 'a limited set'

File: docs/FAQ.md [hunk 2]
```diff
@@ -13,7 +13,174 @@ settings. This is true of all configs in Spark. If you changed
...
-like with `collect`, `show` or `write` a new `DataFrame` is constructed causing spark to replan the
+like with `collect`, `show` or `write` a new `DataFrame` is constructed causing spark to re-plan the
 query. This is why `spark.rapids.sql.enabled` is still respected when running, even if explain
 shows stale results.
 
+### What versions of Apache Spark does the RAPIDS Accelerator for Apache Spark support?
+
+The RAPIDS Accelerator for Apache Spark requires version 3.0.0 of Apache Spark. Because the plugin
+replaces parts of the physical plan that Apache Spark considers to be internal the code for those
+plans can change even between bug fix releases. As a part of our process we try to stay on top of
```
Comment (kuhushukla) [inline docs/FAQ.md:24]: Nit. Comma after process

File: docs/FAQ.md [hunk 3]
```diff
@@ -13,7 +13,174 @@ settings. This is true of all configs in Spark. If you changed
...
+can change from one distribution to another. We are working with most cloud service providers to
+set up testing and validation on their distributions.
+
+### What is the right hardware setup to run GPU accelerated Spark?
+
+Reference Architectures should be available around Q4 2020.
+
+### What CUDA versions are supported?
+
+CUDA 10.1 and 10.2 are currently supported, but you need to download the cudf jar that corresponds
```
Comment (jlowe) [inline docs/FAQ.md:44]: Worth putting a link here, either to the cudf repository or the documentation that discusses how to download cudf?

File: docs/FAQ.md (hunk 1 above)
Comment (jlowe) [inline docs/FAQ.md:49]:
```suggestion
Currently a limited set of SQL and DataFrame operations are supported, please see the
```

File: docs/FAQ.md [hunk 4]
```diff
@@ -13,7 +13,174 @@ settings. This is true of all configs in Spark. If you changed
...
+### What parts of Apache Spark are accelerated?
+
+Currently, A limited set of SQL and DataFrame operations are supported, please see the
+[configs](configs.md) for a more complete list of what is supported. Some of structured streaming
+is likely to be accelerated, but we have not been an area of focus right now. Other areas like
+MLLib, GraphX or RDDs are not accelerated.
+
+### What is the road-map like?
+
+Please take a look at the github repository https://github.com/nvidia/spark-rapids The have issue
```
Comment (kuhushukla) [inline docs/FAQ.md:56]: May need to reword this a bit

File: docs/FAQ.md [hunk 5]
```diff
@@ -13,7 +13,174 @@ settings. This is true of all configs in Spark. If you changed
...
+### What CUDA versions are supported?
+
+CUDA 10.1 and 10.2 are currently supported, but you need to download the cudf jar that corresponds
+to the version you are using.
+
+### What parts of Apache Spark are accelerated?
+
+Currently, A limited set of SQL and DataFrame operations are supported, please see the
+[configs](configs.md) for a more complete list of what is supported. Some of structured streaming
+is likely to be accelerated, but we have not been an area of focus right now. Other areas like
```
Comment (jlowe) [inline docs/FAQ.md:51]:
```suggestion
is likely to be accelerated, but it has not been an area of focus right now. Other areas like
```

File: docs/FAQ.md (hunk 4 above)
Comment (jlowe) [inline docs/FAQ.md:56]:
```suggestion
Please take a look at the [github repository](https://github.com/nvidia/spark-rapids) which has issue
```

File: docs/FAQ.md [hunk 6]
```diff
@@ -13,7 +13,174 @@ settings. This is true of all configs in Spark. If you changed
...
+
+### Is pyspark supported?
+
+Yes
+
+### Are the R APIs for Spark supported?
+
+Yes, but we don't actively test them.
+
+## Are the java APIs for Spark supported?
```
Comment (jlowe) [inline docs/FAQ.md:137]:
```suggestion
## Are the Java APIs for Spark supported?
```

File: docs/FAQ.md [hunk 7]
```diff
@@ -13,7 +13,174 @@ settings. This is true of all configs in Spark. If you changed
...
+possibly in a different format, like CSV to parquet)
+
+### Are there initialization costs?
+
+From our tests the GPU typically takes about 2 to 3 seconds to initialize when an executor first
+starts. If you are only going to run a single query that only takes a few seconds to run this can
+be problematic. In general if you are going to do 30 seconds or more of processing within a single
+session the overhead can be amortized.
+
+### Why is the size of my output Parquet/Orc file different?
```
Comment (abellina) [inline docs/FAQ.md:86]:
```suggestion
### Why is the size of my output Parquet/ORC file different?
```

File: docs/FAQ.md [hunk 8]
```diff
@@ -13,7 +13,174 @@ settings. This is true of all configs in Spark. If you changed
...
+
+## Is the GPU needed on the driver?  Are there any benefits to having a GPU on the driver?
+
+The GPU is not needed on the driver and there is no benefit to having one available on the driver
+for the RAPIDS plugin.
+
+## How does the performance compare to DataBricks' DeltaEngine?
+
+We have not evaluated the performance yet. DeltaEngine is not open source, so any analysis needs to
+be done with Databricks in some form. When DeltaEngine is generally available if the terms of
```
Comment (jlowe) [inline docs/FAQ.md:153]:
```suggestion
be done with Databricks in some form. When DeltaEngine is generally available and the terms of
```

File: docs/FAQ.md [hunk 9]
```diff
@@ -13,7 +13,174 @@ settings. This is true of all configs in Spark. If you changed
...
+service allow it we will look into doing a comparison.
```
Comment (abellina) [inline docs/FAQ.md:154]:
```suggestion
service allow it, we will look into doing a comparison.
```

File: docs/FAQ.md [hunk 10]
```diff
@@ -13,7 +13,174 @@ settings. This is true of all configs in Spark. If you changed
...
+
+### How many tasks can I run per executor? How many should I run per executor?
+
+There is no limit on the number of tasks per executor that you can run.  Generally we recommend 2 to
+6 tasks per executor and 1 GPU per executor. The GPU typically benefits from having 2 tasks run
+in (parallel)[configs.md#sql.concurrentGpuTasks] on it at a time, assuming your GPU has enough
```
Comment (abellina) [inline docs/FAQ.md:160]: This link isn't rendering for some reason (the `parallel` one)

**Comment by revans2** [comment]
build

**Comment by revans2** [comment]
Thanks for the reviews I think I have addressed all of the comments.

**Comment by revans2** [comment]
build

**Pull request merged**